src/Microsoft.Azure.WebJobs.Host/FunctionInvocationContext.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Azure.WebJobs.Host.Executors;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Host
+{
+    /// <summary>
+    /// The context for an executed function. This needs to be expanded on later.
+    /// </summary>
+    public abstract class FunctionInvocationContext
```

Comment:
I don't see that you've updated PublicSurfaceTests.cs with all the new types you're adding. You'll need to run all the tests and get a clean run.